float32 setpoint
float32 time
float32 num_segs


